scipy/signal/_peak_finding.py

```diff
@@ -233,6 +236,733 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
+        A signal with peaks.
+    height : number or ndarray or sequence, optional
+        Required height of peaks. Either a number, ``None``, an array matching
+        `x` or a 2-element sequence of the former. The first element is
+        always interpreted as the  minimum and the second, if supplied, as the
+        maximum required height.
+    threshold : number or ndarray or sequence, optional
+        Required threshold of peaks, the vertical distance to its neighbouring
+        samples. Either a number, ``None``, an array matching `x` or a
+        2-element sequence of the former. The first element is always
+        interpreted as the  minimum and the second, if supplied, as the maximum
+        required threshold.
+    distance : number, optional
+        Required minimal horizontal distance (>= 1) in samples between
+        neighbouring peaks. The removal order is explained in the notes section.
+    prominence : number or ndarray or sequence, optional
+        Required prominence of peaks. Either a number, ``None``, an array
+        matching `x` or a 2-element sequence of the former. The first
+        element is always interpreted as the  minimum and the second, if
+        supplied, as the maximum required prominence.
+    width : number or ndarray or sequence, optional
+        Required width of peaks. Either a number, ``None``, an array
+        matching `x` or a 2-element sequence of the former. The first
+        element is always interpreted as the  minimum and the second, if
+        supplied, as the maximum required prominence.
+    wlen : number, optional
+        Used for calculation of the peaks prominences thus it is only used if
+        one of the arguments `prominence` or `width` is given. See argument
+        `wlen` in `peak_prominences` for a full description of its effects.
+    rel_height : float, optional
+        Used for calculation of the peaks width thus it is only used if `width`
+        is given. See argument  `rel_height` in `peak_widths` for a full
+        description of its effects.
+
+    Returns
+    -------
+    peaks : ndarray
+        Indices of peaks in `x` that satisfy all requirements.
+    properties : dict
+        A dictionary containing properties of the returned peaks which were
+        calculated as intermediate results during filtering:
+
+        * If `height` is given, the value of `x` at each peak is returned
+          with the key 'peak_heights'.
+        * If `threshold` is given, the keys 'left_thresholds' and
+          'right_thresholds' contain a peaks vertical distance to its
+          neighbouring samples.
+        * If `prominence` is given, the keys 'prominences', 'left_bases' and
+          'right_bases' are accessible. See `peak_prominences` for a description
+          of their content.
+        * If `width` is given, the keys 'wheights', 'left_ips' and 'right_ips'
+          are accessible. See `peak_widths` for a description of their content.
+
```

Comment:
"This will be an empty dictionary if no peaks are found"

Or maybe better, don't make it empty (as happens now), but instead populate the proper keys with empty arrays